src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala

```diff
@@ -13,28 +13,42 @@ import org.json4s.JsonDSL._
 object EthJsonMethodsImplicits extends JsonMethodsImplicits {
 
   implicit val eth_protocolVersion = new JsonDecoder[ProtocolVersionRequest] with JsonEncoder[ProtocolVersionResponse] {
-    def decodeJson(params: Option[JArray]): Either[JsonRpcError, ProtocolVersionRequest] = Right(ProtocolVersionRequest())
+    def decodeJson(params: Option[JArray]): Either[JsonRpcError, ProtocolVersionRequest] = Right(
+      ProtocolVersionRequest()
+    )
 
     def encodeJson(t: ProtocolVersionResponse): JValue = t.value
   }
 
+  implicit val eth_chainId = new JsonDecoder[ChainIdRequest] with JsonEncoder[ChainIdResponse] {
+    def decodeJson(params: Option[JArray]) = params match {
+      case None | Some(JArray(Nil)) => Right(ChainIdRequest())
+      case other => Left(InvalidParams("eth_chainId method does not take params"))
```

Comment:
I realized that other parameterless methods don't check if `params` is empty. Maybe we could improve that in this PR. WDYT?

Reply:
If only @pslaski agrees (this will result in quite huge PR due to formatting)

Reply:
this PR is already huge so go for it ;)